### PR TITLE
move array_replace family functions to datafusion-function-array crate

### PR DESCRIPTION
**Pull request by Weijun-H**
## Which issue does this PR close?

Relate #9285

### CODE REVIEW

**Review by comphead — APPROVED** [review]
lgtm thanks @Weijun-H

File: datafusion/functions-array/src/kernels.rs [hunk 1]
```diff
@@ -1207,3 +1207,104 @@ where
...
+/// ```text
+/// compare_element_to_list(
+///     [1, 2, 3], [1, 2, 3], 0, true => [true, false, false]
+///     [1, 2, 3, 3, 2, 1], [1, 2, 3], 1, true => [false, true, false, false, true, false]
+///
+///     [[1, 2, 3], [2, 3, 4], [3, 4, 5]], [[1, 2, 3], [2, 3, 4], [3, 4, 5]], 0, true => [true, false, false]
+///     [[1, 2, 3], [2, 3, 4], [2, 3, 4]], [[1, 2, 3], [2, 3, 4], [3, 4, 5]], 1, false => [true, false, false]
+/// )
+/// ```
+pub(crate) fn compare_element_to_list(
```
Comment (jayzhan211) [inline datafusion/functions-array/src/kernels.rs:1238]: We can move it to `utils`, not only replace function relies on it.

File: datafusion/functions-array/src/lib.rs [hunk 2]
```diff
@@ -115,6 +119,9 @@ pub fn register_all(registry: &mut dyn FunctionRegistry) -> Result<()> {
         set_ops::array_union_udf(),
         position::array_position_udf(),
         position::array_positions_udf(),
+        array_replace::array_replace_n_udf(),
```
Comment (jayzhan211) [inline datafusion/functions-array/src/lib.rs:122]: nit: I think we can name it replace

File: datafusion/core/benches/array_expression.rs [hunk 3]
```diff
@@ -22,48 +22,32 @@ extern crate datafusion;
...
-                array_expressions::array_replace_all(args.as_slice())
-                    .unwrap()
-                    .as_list::<i32>(),
-                criterion::black_box(&expected_array)
+                array_replace_all(
+                    list_array.clone(),
+                    from_array.clone(),
+                    to_array.clone()
+                ),
+                criterion::black_box(&expected_array).clone()
```
Comment (jayzhan211) [inline datafusion/core/benches/array_expression.rs:50]: is it possible to avoid the clone here? it is already cloned in L38

**Pull request merged**